guides.Rmd

```diff
@@ -89,78 +120,168 @@ axs + scale_x_continuous(breaks = c(2000, 4000))
+\index{Minor breaks}\index{Log!ticks}
+```{r}
+mb <- as.numeric(1:10 %o% 10 ^ (0:4))
+```
+The following plots illustrate the effect of setting the minor breaks:
 `r columns(2, 2/3)`
 ```{r}
-df <- data.frame(x = c(2, 3, 5, 10, 200, 3000), y = 1)
-ggplot(df, aes(x, y)) + 
-  geom_point() + 
-  scale_x_log10()
+base <- ggplot(toy, aes(log, const)) + geom_point()
 
-mb <- as.numeric(1:10 %o% 10 ^ (0:4))
-ggplot(df, aes(x, y)) + 
-  geom_point() + 
-  scale_x_log10(minor_breaks = mb)
+base + scale_x_log10()
+base + scale_x_log10(minor_breaks = mb)
+```
+ 
+Note also that the scales package contains `minor_breaks_n()` and `minor_breaks_width()` functions that can be helpful in controlling the minor breaks.
+
+
+### Date scales {#date-scales}
+
+A special case of interest occurs when an aesthetic is mapped to date or date/time data. In ggplot2 these are treated as continuous variables that have special labels, and there are some helper functions designed to make it easier to specify the breaks and labels for these scales. The ggplot2 package supports the `Date` (for dates) and `POSIXct` (for date/times) classes, as well as the `hms` class provided by the hms package [@hms]. If your dates are in a different format you will need to convert them using `as.Date()`, `as.POSIXct()` or `hms::as_hms()`. You may also find the toolkit provided by the lubridate package helpful to manipulate time and date data [@lubridate].
+
+
+For appropriately formatted data mapped to the x aesthetic, ggplot2 will default to `scale_x_date()` and `scale_x_datetime()`, and there are similar scales for other aesthetics. In most respects date scales behave the same way as other continuous scales, but have some special behaviour that is worth noting. Specifically, in addition to the usual `breaks` and `labels` arguments, date scales have special `date_breaks` and `date_labels` arguments that allow you to work in date-friendly units:
+\index{Date/times} \index{Data!date/time} \index{Time} \index{Scales!date/time} \indexf{scale\_x\_datetime}
+
+*   `date_breaks` and `date_minor_breaks()` allow you to position breaks by
+    date units (years, months, weeks, days, hours, minutes, and seconds). 
+    For example, `date_breaks = "2 weeks"` will place a major 
+    tick mark every two weeks.
+
+*   `date_labels` controls the display of the labels
+    using the same formatting strings as in `strptime()` and `format()`. 
+    To display dates like 14/10/1979, for example, you would use the 
+    string `"%d/%m/%Y"`: in this expression `%d` produces a numeric 
+    day of month, `%m` produces a numeric month, and `%Y` produces a 
+    four digit year (the table at the end of this section provides a 
+    longer list of formatting strings)
```

Comment:
Wonder if it's worth explicitly mentioning `"\n"` here? Then you can have (e.g.) month and year on separate lines.